tests/Makefile

```diff
@@ -269,8 +269,8 @@ endif
 
 # Test suites caught by SKIP_TEST_SUITES are built but not executed.
 check: $(BINARIES) $(CRYPTO_BINARIES)
-	perl scripts/run-test-suites.pl $(TEST_FLAGS) --skip=$(SKIP_TEST_SUITES)
 	cd ../tf-psa-crypto/tests && perl ../../tests/scripts/run-test-suites.pl $(TEST_FLAGS) --skip=$(SKIP_TEST_SUITES)
+	perl scripts/run-test-suites.pl $(TEST_FLAGS) --skip=$(SKIP_TEST_SUITES)
```

Comment:
Yeah, I thought about it. I guess the easiest way to always run all the tests would be:
```
(cd ../tf-psa-crypto/tests && perl ../../tests/scripts/run-test-suites.pl $(TEST_FLAGS) --skip=$(SKIP_TEST_SUITES)); perl scripts/run-test-suites.pl $(TEST_FLAGS) --skip=$(SKIP_TEST_SUITES)
```

Reply:
Ah no, silly me, that would exit 0 if a crypto test fails but for some reason the X.509/TLS part still succeeds.

Reply:
The thing is, I wanted to first take the easy win (better ordering) without necessarily having to do the harder part (always running them all). Especially considering this is likely temporary, as I assume once tf-psa-crypto is its own repo with its own CI, we will no longer be running its test suites in Mbed TLS because we'll only update the submodule to commits that already got a green CI on the tf-psa-crypto side. (And also, there were discussion of abandoning make, though I'm not sure that's happening soon considering it would require non-trivial work on cmake.)

Reply:
The shell code can be fixed, but if we invoke `run-test-suites.pl` multiple times, the last line will be wrong since it will only count the test suites from the last run of the script.

Reply:
Indeed, but that's a pre-existing issue.

Reply:
Both the splitting of the summary and the conditional partial skipping come from invoking `run-test-suites.pl` twice.